nncf/openvino/engine.py

```diff
@@ -0,0 +1,188 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from typing import Callable
+from time import time
+
+from openvino.runtime import AsyncInferQueue
+from openvino.tools import pot
+
+from nncf.api.compression import ModelType
+from nncf.data import DataLoader
+from nncf.data.utils import create_subset
+
+
+logger = pot.utils.logger.get_logger(__name__)
+
+
+class DummyMetric(pot.Metric):
+    def __init__(self, higher_better: bool = True):
+        self._name = 'custom_metric'
+        self._higher_better = higher_better
+        self._avg_value = None
+
+    @property
+    def higher_better(self):
+        return self._higher_better
+
+    @property
+    def avg_value(self):
+        return {self._name: self._avg_value}
+
+    @avg_value.setter
+    def avg_value(self, value):
+        self._avg_value = value
+
+    @property
+    def value(self):
+        raise NotImplementedError()
+
+    def get_attributes(self):
+        attributes = {
+            self._name: {
+                'direction': 'higher-better' if self.higher_better else 'higher-worse',
+                'type': 'user_type',
+            }
+        }
+        return attributes
+
+    def update(self, output, target):
+        raise NotImplementedError()
+
+    def reset(self):
+        self._avg_value = None
+
+
+def _pot_sampler_to_nncf_dataloader(sampler) -> DataLoader:
+    dataloader = sampler._data_loader
+    indices = sampler._subset_indices
+
+    if not isinstance(dataloader, DataLoader):
+        raise Exception(f'Unexpected type of data loader: {type(dataloader)}')
+
+    if isinstance(sampler, pot.samplers.batch_sampler.BatchSampler):
+        return create_subset(dataloader, indices)
+
+    raise Exception(f'Unexpected type of sampler: {type(sampler)}')
+
+
+class CustomEngine(pot.IEEngine):
```

Comment:
I would call it `OVEngine`, `NNCFOVEngine` or whatever. `Custom` is too generic.

Reply:
That's a good point. I renamed it.